src/users/users.controller.ts

```diff
@@ -119,17 +122,27 @@ export class UsersController {
         description: "Given email is not a valid email.",
         type: BadRequestErrorResponse,
     })
-    @ApiParam({
-        name: "email",
-        required: true,
-        description: "email",
-        example: "jessica.williamson@gmail.com",
+    @ApiBody({
+        schema: {
+            type: "object",
+            properties: {
+                email: {
+                    type: "string",
+                    description: "email",
+                },
+            },
+        },
     })
     @Roles(AppRoles.Admin)
-    @Get("email/:email")
-    getUserDetailsByEmail(@Param("email") email: string) {
+    @Post("/lookup-by-email")
+    async getUserDetailsByEmail(@Body("email") email: string) {
```

Comment:
should we use a DTO with email validation instead? https://docs.nestjs.com/techniques/validation#auto-validation

Could use something like this 
```
export class UserLookupByEmailDto {   
  @IsEmail()
  email: string;
}
```

Reply:
ok, I think this will be better. I was unsure of how to go about it.